core/corehttp/proxy_test.go

```diff
@@ -0,0 +1,34 @@
+package corehttp
+
+import (
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/ipfs/go-ipfs/thirdparty/assert"
+)
+
+func TestParseRequest(t *testing.T) {
+	url := "http://localhost:5001/p2p/QmT8JtU54XSmC38xSb1XHFSMm775VuTeajg7LWWWTAwzxT/http/path/to/index.txt"
```

Comment:
We need to add at least one more test-case here for the `/x/custom/http` case. I would do it so that it can easily be expanded to the `/ws` cases in the future (loop on a slice of struct containing testcases strings and expected results for each).

Reply:
I've added more test cases